src/api/custom-schema.ts

```diff
@@ -39,5 +72,40 @@ export const extendGraphqlSchema = graphQLSchemaExtension<Context>({
         }
       },
     },
+    Query: {
+      // call for companies house API
+      getCompaniesHouseInformation: async (root, variables) => {
+        try {
+          const { companiesHouseNumber } = variables;
+          console.info('Calling Companies House API for ', companiesHouseNumber);
+          const sanitisedRegNo: number = companiesHouseNumber.padStart(8, '0');
+
+          const response = await axios({
+            method: 'get',
+            url: `${companiesHouseURL}/company/${sanitisedRegNo}`,
+            auth: { username, password: '' },
+            validateStatus(status) {
+              return status === 200 || status === 404;
+            },
+          });
+
+          // if no data in response or status is not 200 then return blank object
+          if (!response.data || response.status === 404) {
+            return {
+              success: false,
+            };
+          }
+
+          // maps response to camelCase fields
+          const mappedResponse = mapCompaniesHouseFields(response.data);
+          return mappedResponse;
+        } catch (error) {
+          console.error('Error calling Companies House API', { error });
+          return {
+            apiError: true,
+          };
```

Comment:
is `apiError` for the requirment to tell if companies house API is down?

Should we add `success: false` as well?

Reply:
adding success false will cause multiple errors which is unwanted, this way will be easier to manage.  And yes apiError is for if the API is down for example

Reply:
done